src/hooks/useJoinSponsorship.ts

```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react'
+import { toaster } from 'toasterhea'
+import { useWalletAccount } from '~/shared/stores/wallet'
+import { useMyOperator } from '~/hooks/useMyOperator'
+import JoinSponsorshipModal from '~/modals/JoinSponsorshipModal'
+import { Layer } from '~/utils/Layer'
+import getSponsorshipTokenInfo from '~/getters/getSponsorshipTokenInfo'
+import { stakeOnSponsorship } from '~/services/sponsorships'
+import { toBN } from '~/utils/bn'
+
+const joinSponsorshipModal = toaster(JoinSponsorshipModal, Layer.Modal)
+
+export const useJoinSponsorship = (): {
+    canJoinSponsorship: boolean
+    joinSponsorship: (sponsorshipId: string, sponsorshipStreamId: string) => Promise<void>
+} => {
+    const wallet = useWalletAccount()
+    const myOperatorQuery = useMyOperator(wallet || '')
+    const canJoinSponsorship = useMemo(() => !!myOperatorQuery.data, [myOperatorQuery])
+
+    return {
+        canJoinSponsorship,
+        joinSponsorship: async (sponsorshipId, sponsorshipStreamId) => {
+            if (!myOperatorQuery.data) {
+                return
+            }
+            const balance = myOperatorQuery.data?.poolValue.minus(
```

Comment:
We should use `operator.freeFundsWei` to get available balance on operator contract.

Reply:
Wel actually `freeFundsWei` shows invalud value :) it doesn't subtract the staked amount - that's why I did the math here. I'll raise the question about it on today's status call.

Reply:
Hmm, weird. For me it seems to be working. I tried it with operator delegate/undelegate and it shows correctly poolvalue subtracted by amount staked in sponsorships.